packaging/debian/postinst

```diff
@@ -25,31 +25,82 @@ DAEMON_USER=storj-storagenode
+    unzip $tmpfile -d /var/lib/storagenode
+    mv /var/lib/storagenode/${binary}_linux_amd64 /var/lib/storagenode/${binary}
+    rm $tmpfile
+}
 
 case "$1" in
     configure)
 	installed_version="$2"
-	#if [ $installed_version != "" ]; then
-	#    echo "This is a reconfiguration or a reinstallation"
-	#fi
+	if [ -n "$installed_version" ]; then
+	    echo "This is a reconfiguration or a reinstallation"
+	fi
+
+	db_get storagenode/identitydir
+	identitydir="$RET"
+	db_get storagenode/wallet
+	wallet="$RET"
+	db_get storagenode/email
+	email="$RET"
+	db_get storagenode/external_addr
+	external_addr="$RET"
+	db_get storagenode/storagedir
+	storagedir="$RET"
+	db_get storagenode/diskspace
+	diskspace="$RET"
 
-	# TODO: should not download binaries if it's only a reconfigure
-	#echo "Downloading binaries..."
 	for binary in "storagenode" "storagenode-updater"; do
-	    wget -O /tmp/${binary}.zip ${BINARIES_SERVER}/${binary}_linux_amd64.zip
-	    unzip /tmp/${binary}.zip -d /var/storagenode/
-	    mv /var/storagenode/${binary}_linux_amd64 /var/storagenode/${binary}
-	    # so that we're in PATH easily
-	    ln -s /var/storagenode/${binary} /usr/bin/${binary}
+	    if [ ! -f "/var/lib/storagenode/${binary}" ]; then
+	        echo "Downloading ${binary} binary..."
+	        get_binary $binary
+	    fi
+
+	    if [ ! -f "/usr/bin/${binary}" ]; then
+			ln -s /var/lib/storagenode/${binary} /usr/bin/${binary}
+	    fi
 	done
-	adduser --system --group --no-create-home --quiet $DAEMON_USER
 
-#	if [ "$old_storagedir" != "" ] && [ "$storagedir" != "$old_storagedir" ]; then
-#	    echo "New storage directory: ${storagedir} (was ${old_storagedir})"
-#	else
-#	    echo "Storage directory: ${storagedir}"
-#	fi
+	# TODO: handle storage dir change during reconfigure
+	if [ -n "$old_storagedir" ] && [ "$storagedir" != "$old_storagedir" ]; then
+	    echo "New storage directory: ${storagedir} (was ${old_storagedir})"
+	else
+	    echo "Storage directory: ${storagedir}"
+	fi
+
+	# copy identity
+	cp ${identitydir}/identity.cert /etc/storagenode/identity
+	cp ${identitydir}/identity.key /etc/storagenode/identity
+
+	# storagenode setup
+	if [ ! -f "/etc/storagenode/config.yaml" ]; then
+	    /var/lib/storagenode/storagenode setup \
+	    --config-dir "/etc/storagenode" \
+	    --identity-dir "/etc/storagenode/identity" \
+	    --operator.email "$email" \
+	    --operator.wallet "$wallet" \
+	    --contact.external-address "$external_addr" \
+	    --storage.path "$storagedir" \
+	    --storage.allocated-disk-space "$diskspace"
+
+	    # fix permissions
+	    chmod 644 /etc/storagenode/config.yaml
+	fi
 	
+	# add user and fix ownership
+	if ! getent passwd $DAEMON_USER >/dev/null; then
+	    adduser --system --group --disabled-login --home /var/lib/storagenode --quiet $DAEMON_USER
+	fi
+
+	chown -R $DAEMON_USER:$DAEMON_USER /var/lib/storagenode
+	chown -R $DAEMON_USER:$DAEMON_USER /etc/storagenode
     ;;
 
     abort-upgrade|abort-remove|abort-deconfigure)
```

Comment:
And this line can just be `abort-deconfigure)` (we're moving `abort-upgrade` and `abort-remove` up with configure).  We don't need to do anything special in the abort-deconfigure case.